native/src/routes/feedback/containers/FeedbackModalContainer.js

```diff
@@ -1,33 +1,287 @@
+    }
+    return CityModel.findCityName(cities, cityCode)
+  }
+
+  getFeedbackOptions = (): Array<FeedbackVariant> => {
+    const options = []
+    const currentPageFeedbackOption = this.getCurrentPageFeedbackOption()
+    if (currentPageFeedbackOption) {
+      options.push(currentPageFeedbackOption)
+    }
+
+    const contentFeedbackOption = this.getContentFeedbackOption()
+    if (contentFeedbackOption) {
+      options.push(contentFeedbackOption)
+    }
+
+    options.push(...this.getOffersFeedbackOptions(), this.getTechnicalFeedbackVariant())
+
+    return options
+  }
+
+  getTechnicalFeedbackVariant = () => {
+    return new FeedbackVariant({
+      label: this.props.t('technicalTopics'),
+      feedbackType: CATEGORIES_FEEDBACK_TYPE,
+      feedbackCategory: TECHNICAL_FEEDBACK_CATEGORY
+    })
+  }
+
+  getContentFeedbackOption = (): ?FeedbackVariant => {
+    const { navigation, t } = this.props
+    const cityName = this.getCityName()
+
+    const label = t('contentOfCity', { city: cityName })
+    const feedbackCategory = CONTENT_FEEDBACK_CATEGORY
+    const feedbackType = navigation.getParam('type')
+    switch (feedbackType) {
+      case 'Event':
+        return new FeedbackVariant({ label, feedbackType: EVENTS_FEEDBACK_TYPE, feedbackCategory })
+      case 'Offers':
+        return new FeedbackVariant({ label, feedbackType: OFFERS_FEEDBACK_TYPE, feedbackCategory })
+      default:
+        return new FeedbackVariant({ label, feedbackType: CATEGORIES_FEEDBACK_TYPE, feedbackCategory })
+    }
+  }
+
+  getOffersFeedbackOptions = (): Array<FeedbackVariant> => {
+    const { t } = this.props
+    const { offers } = this.props.navigation.state.params
+    if (offers) {
+      return offers.map(offer =>
+        new FeedbackVariant({
+          label: t('contentOfOffer', { offer: offer.title }),
+          feedbackType: OFFER_FEEDBACK_TYPE,
+          feedbackCategory: CONTENT_FEEDBACK_CATEGORY,
+          alias: offer.alias
+        })
+      )
+    }
+    return []
+  }
 
-class FeedbackModalContainer extends React.Component<{| navigation: NavigationStackProp<*> |}> {
   closeModal = () => this.props.navigation.goBack()
 
-  sendFeedback = async (feedbackData: FeedbackParamsType) => {
-    const apiUrl = await determineApiUrl()
-    const feedbackEndpoint = createFeedbackEndpoint(apiUrl)
-    await feedbackEndpoint.request(feedbackData)
+  getCurrentPageFeedbackOption = (): ?FeedbackVariant => {
+    const { navigation, t } = this.props
+    const { type, path, title, alias } = navigation.state.params
+
+    const feedbackCategory = CONTENT_FEEDBACK_CATEGORY
+    if (type === 'Category' && path && title) {
+      return new FeedbackVariant({
+        label: t('contentOfPage', { page: title }),
+        feedbackType: PAGE_FEEDBACK_TYPE,
+        feedbackCategory
+      })
+    } else if (type === 'Pois' && path && title) {
+      return new FeedbackVariant({
+        label: t('contentOfPoi', { poi: title }),
```

Comment:
This is also related to to IGAPP-21. Which is about creating the translations for the pois page.

Reply:
I think this would normally be part of IGAPP-337. You can add the key with a dummy translation already if you want though. We don't know what to call POIs yet though afaik.

Reply:
I'd suggest to just add it here to the translations as it is just one line. That way we do not forget it.